### DISCUSSION TITLE
New UI?

### DISCUSSION THREAD

**Discussion by dj-make** [twinbasic/twinbasic#1747] [General]
Fan-modified UI or official new UI?

![95f7d7ca7bcb0a4640640d0e2d63f6246b60af4c](https://github.com/twinbasic/twinbasic/assets/122567615/d260a3c7-f3a4-4298-9b18-91150b488220)

**Comment by woeoio** [comment]
official new UI  in next version.

**Comment by WaynePhillipsEA** [comment]
Yes, this is the new look UI coming in the next release

**Reply by dj-make** [reply]
> Yes, this is the new look UI coming in the next release

Can you design the window below in tab mode?
![QQ截图20240703110525](https://github.com/twinbasic/twinbasic/assets/122567615/02521e6c-7520-4ccf-8403-9653a9aa76f6)
![QQ截图20240703110427](https://github.com/twinbasic/twinbasic/assets/122567615/3fdf3238-efd3-4bd8-ad4e-df1b77953458)
![QQ截图20240703110623](https://github.com/twinbasic/twinbasic/assets/122567615/98575e3e-5f09-468d-becf-22e35f6e3744)

**Reply by dj-make** [reply]
> 是的，这是下一个版本中的新外观 UI

It is recommended to use the following modes when editing multiple windows, otherwise it will waste the display area too much.
![QQ截图20240703115125](https://github.com/twinbasic/twinbasic/assets/122567615/a327c6b9-3e1e-41fd-bd50-5f63b0486df4)
![QQ截图20240703115050](https://github.com/twinbasic/twinbasic/assets/122567615/4eb6b820-8c86-4673-aa34-d769e207fdeb)

**Reply by fafalone** [reply]
I don;t like the idea of needing to scroll to find open windows, but those tabs kinda do need to be shrunk.

**Comment by ElHado** [comment]
@WaynePhillipsEA Great, I love it 👍

**Comment by dmrvb** [comment]
A few UI observations/requests (that might not be possible)

1 When a prog is running the GUI is faded, including the debug console. Reading Debug.Prints is part of software development so ideally the debug.console should not fade.

2 Clicking the top and bottom scroll bar buttons should ideally scroll the display by exactly one line of text, this would be of most use in code window.

3 I would love to see the VB6 option of displaying only a single Function/Sub in the code window. This is especially useful when handling big modules with lots of subs/functions.

4 I am currently loading the ucShellBrowse Twinpack. This takes a little time to load. When handling any Twinpack references there is some time delay and no visual feedback that anything is happening. This was confusing the first couple of times that I did it. A progress bar, or message window, or "egg timer/busy" mousepointer would be nice.
